code/ship/ship.cpp

```diff
@@ -2281,6 +2204,136 @@ static int parse_and_add_briefing_icon_info()
+{
+	Assert(info_type_name != nullptr);
+	Assert(info_name != nullptr);
+
+	// for parsing
+	const char *prefix = (direction == WarpDirection::WARP_IN) ? "$Warpin" : "$Warpout";
+	char str[NAME_LENGTH];
+
+	WarpParams params;
+	if (inherit_from != nullptr)
+		params = *inherit_from;
+	params.direction = direction;
+
+	sprintf(str, "%s type:", prefix);
+	if (optional_string(str))
+	{
+		char buf[NAME_LENGTH];
+
+		stuff_string(buf, F_NAME, NAME_LENGTH);
+		int j = warptype_match(buf);
+		if (j >= 0) {
+			params.warp_type = j;
+		}
+		else {
+			// try to match the warp type with one of our fireballs
+			j = fireball_info_lookup(buf);
+			if (j >= 0) {
+				params.warp_type = j | WT_DEFAULT_WITH_FIREBALL;
+			}
+			else {
+				Warning(LOCATION, "Invalid %s '%s' specified for %s '%s'", str, buf, info_type_name, info_name);
+				params.warp_type = WT_DEFAULT;
+			}
+		}
+	}
+
+	sprintf(str, "%s Start Sound:", prefix);
+	parse_game_sound(str, &params.snd_start);
+
+	sprintf(str, "%s End Sound:", prefix);
+	parse_game_sound(str, &params.snd_end);
+
+	if (direction == WarpDirection::WARP_OUT)
+	{
+		sprintf(str, "%s engage time:", prefix);
+		if (optional_string(str))
+		{
+			float t_time;
+			stuff_float(&t_time);
+			if (t_time > 0.0f)
+				params.warpout_engage_time = fl2i(t_time*1000.0f);
+			else
+				Warning(LOCATION, "%s specified as 0 or less on %s '%s'; value ignored", str, info_type_name, info_name);
+		}
+	}
+
+	sprintf(str, "%s speed:", prefix);
+	if (optional_string(str))
+	{
+		float speed;
+		stuff_float(&speed);
+		if (speed > 0.0f)
+			params.speed = speed;
+		else
+			Warning(LOCATION, "%s specified as 0 or less on %s '%s'; value ignored", str, info_type_name, info_name);
+	}
+
+	sprintf(str, "%s time:", prefix);
+	if (optional_string(str))
+	{
+		float t_time;
+		stuff_float(&t_time);
+		if (t_time > 0.0f)
+			params.time = fl2i(t_time*1000.0f);
+		else
+			Warning(LOCATION, "%s specified as 0 or less on %s '%s'; value ignored", str, info_type_name, info_name);
+	}
+
+	sprintf(str, "%s %s exp:", prefix, direction == WarpDirection::WARP_IN ? "decel" : "accel");
+	if (optional_string(str))
+	{
+		float accel_exp;
+		stuff_float(&accel_exp);
+		if (accel_exp >= 0.0f)
+			params.accel_exp = accel_exp;
+		else
+			Warning(LOCATION, "%s specified as less than 0 on %s '%s'; value ignored", str, info_type_name, info_name);
+	}
+
+	sprintf(str, "%s radius:", prefix);
+	if (optional_string(str))
+	{
+		float rad;
+		stuff_float(&rad);
+		if (rad > 0.0f)
+			params.radius = rad;
+		else
+			Warning(LOCATION, "%s specified as 0 or less on %s '%s'; value ignored", str, info_type_name, info_name);
+	}
+
+	sprintf(str, "%s animation:", prefix);
+	if (optional_string(str))
+	{
+		stuff_string(params.anim, F_NAME, MAX_FILENAME_LEN);
+	}
+
+	if (direction == WarpDirection::WARP_OUT)
+	{
+		sprintf(str, "$Player warpout speed:");
+		if (optional_string(str))
```

Comment:
This reads awkward as hell; would just using plain text for this one option really be that bad?

Reply:
It's a little silly, but I did that for consistency.  Not just in that one line, but more importantly for the `Warning` a few lines later.  That is the same for each field (except for `accel_exp` which allows 0) and requires the `str` parameter.